src/porepy/applications/md_grids/model_geometries.py

```diff
@@ -196,3 +197,273 @@ def create_mdg(self) -> None:
+        setting boundary conditions or source terms, the user should ensure consistency
+        with these names, and may override this property to provide custom names or
+        switch the roles of the wells.
+
+        """
+        return ["injection_well", "production_well"]
+
+    def set_well_network(self) -> None:
+        """Set the two wells.
+
+        See below comment for a sketch of the setup.
+        """
+        # With constant y coordinates for both wells, the projection in the x-z plane
+        # looks roughly as follows, using double lines to indicate the domain
+        # boundaries:
+
+        #               w1      w2
+        #     ==============================
+        #     ||        |        |         ||
+        #     ||        |        |         ||
+        #     ||        |        |         ||
+        #     ||        |        \         ||
+        #     ||        |         \        ||
+        #     ||        |          \       ||
+        #     ||                           ||
+        #     ==============================
+
+        # Side lengths of the domain:
+        dx, dy, dz = self.domain_sizes()
+        # One straight vertical well at (0.35dx, 0.35dy) extending from z=0 to z=-0.8dz.
+        well_1 = pp.Well(
+            np.array([[0.35 * dx, 0.35 * dx], [0.35 * dy, 0.35 * dy], [0, -0.8 * dz]]),
+            tags={"well_name": self.well_names[0]},
+        )
+        # One kinked well at (0.6dx, 0.65dy), with a kink at z=-0.4dz and terminating at
+        # (0.7dx, 0.65dy, -0.8dz).
+        well_2 = pp.Well(
+            np.array(
+                [
+                    [0.6 * dx, 0.6 * dx, 0.7 * dx],
+                    [0.65 * dy, 0.65 * dy, 0.65 * dy],
+                    [0, -0.4 * dz, -0.8 * dz],
+                ]
+            ),
+            tags={"well_name": self.well_names[1]},
+        )
+        self._wells = [well_1, well_2]
+
+        mesh_size = self.params.get("well_mesh_size", {"mesh_size": 0.1 * dz})
+        self.well_network = pp.WellNetwork3d(
+            domain=self._domain, wells=self._wells, parameters=mesh_size
+        )
+
+
+class TwoEllipticFractures3d(SubsurfaceCuboidDomain):
+    """A mixin adding two elliptic fractures to a 3d model.
+
+    The fractures are defined by their centers, major and minor axes, strike and dip
+    angles, and major axis angles. The parameters can be controlled by passing a
+    dictionary ``fracture_params`` to the model parameter dictionary. See the property
+    :meth:`fracture_params` for details on the available parameters and their default
+    values.
+
+    If extending to more than two fractures, the user should override all properties
+    defining fracture parameters to return arrays of size (at least) self.num_fractures.
+    The case num_fractures < (size of arrays) is allowed, in which case only the first
+    num_fractures entries are used.
+
+    TODO: Decide whether to replace properties with a defualt dictionary, i.e.,
```

Comment:
I lean towards implemetning this. Most of the transparency can be kept by proper documentation in such a unified method.